xlml/apis/task.py

```diff
@@ -73,7 +77,22 @@ def run(self) -> DAGNode:
       run_model = self.run_model(queued_resource, ssh_keys)
       post_process = self.post_process()
       clean_up = self.clean_up(queued_resource)
-      provision >> run_model >> post_process >> clean_up
+
```

Comment:
Instead of having flags here, could we have `run_with_prepend_tasks` and `run_with_append_tasks`, which is similar to `run()` and `run_with_startup_script`? So you could directly call those, instead of passing flags.

Reply:
I forget to ask the use cases for `prepend_tasks` and `append_tasks`. Could you explain a little bit more?

Reply:
The `prepend_tasks` and `append_tasks` are for adding custom tasks to the main test job. This is so that we can add task dependencies on the config level rather than at the dag level. In this use-case, we need to have tasks to generate the `run_name` and the `tb_file_location` and we need to be able to specify it as a dependency to the main test job. I think it is possible to add the tasks on the dag-level however I don't think it is something we should do since the user would be able to specify a `run_name` without using a task.

Reply:
@RissyRan @jonb377 I followed Ran's suggestion for creating a new run function `run_with_run_name_generation()`. I made this change just for the GCE flow so far and if it looks good I will change it for the GKE flow as well.